### PR TITLE
System Font Stacks: update to match GitHub

### PR DESCRIPTION
**Pull request by davidrans**
See some analysis by Federico here: https://github.com/iFixit/react-commerce/pull/626#issuecomment-1222652071

As well as some additional references:
- https://bitsofco.de/the-new-system-font-stack/ (2016 - very specific
  on the -apple-system/BlinkMacSystemFont distinction)
- https://markdotto.com/2018/02/07/github-system-fonts/ (2018)
- https://medium.com/towards-more-beautiful-web-typography/survey-system-font-stack-5f73a3b39776 (2020)
- https://css-tricks.com/snippets/css/system-font-stack/ (2022)
- https://make.wordpress.org/core/2016/07/07/native-fonts-in-4-6/ (2016)
- https://qwtel.com/posts/software/the-monospaced-system-ui-css-font-stack/#fn:10 (2020)

Merge checklist
---

- [ ] Bump the package version by running `npm version [patch | minor | major]` from the command line (if applicable).

> **Note:** In the context of Core Primitives, significant changes to the library or workflow, or removing primitives would be considered a major update, adding or updating primitives would be considered a minor update, and fixing primitives would be considered a patch. Non-code changes (e.g. documentation) do not require a version bump.

### CODE REVIEW

**Comment by ianrohde** [comment]
Was `system-ui` considered instead of `-apple-system` and `BlinkMacSystemFont`?

```
Glyphs are taken from the default user interface font on a given platform. Because typographic traditions vary widely across the world, this generic is provided for typefaces that don't map cleanly into the other generics.
```

https://developer.mozilla.org/en-US/docs/Web/CSS/font-family#system-ui
https://caniuse.com/font-family-system-ui

I still have nightmares of tracking down that Chrome issue with `BlinkMacSystemFont`. 😆

**Comment by ianrohde** [comment]
https://css-tricks.com/snippets/css/system-font-stack/ confirms:

```
More recently, Chrome and Sadari shipped a system-ui, which is a generic font family that can replace -apple-system and BlinkMacSystemFont. That means the GitHub snippet could be reduced to this:

/* System Fonts with system-ui */
body {
  font-family: system-ui, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
}
```

**Comment by davidrans** [comment]
Seems like there might be some internationalization issues with `system-ui`?
https://infinnie.github.io/blog/2017/systemui.html
https://github.com/twbs/bootstrap/pull/22377

**Comment by ianrohde** [comment]
We're going to find all sorts of conflicting opinions. I'll spend some time today testing. IIRC have a Windows VM installed on one of my computers

**Comment by ianrohde** [comment]
Ughhhh more Blink issues that were patched https://bugs.chromium.org/p/chromium/issues/detail?id=1018581

File: index.json
Comment (jarstelfox) [inline index.json:244]: @davidrans, see https://github.com/iFixit/ifixit/pull/37335, https://github.com/iFixit/core-primitives/pull/21, and https://bugs.chromium.org/p/chromium/issues/detail?id=781344&q=BlinkMacSystemFont&can=1&sort=-modified

In short, `BlinkMacSystemFont` causes a whole page reflow hurting LCP, FID and CLS of course only in chrome

**Comment by jarstelfox** [comment]
@ianrohde I think you are looking for https://bugs.chromium.org/p/chromium/issues/detail?id=781344&q=BlinkMacSystemFont&can=1&sort=-modified

>Generic font families lead to a reflow performance hit

CC @federicobadini

File: index.json
Comment (ianrohde) [reply index.json:244]: No good then, that's ~75% of traffic and all those metrics are important to rank, UX, etc

**Comment by ianrohde** [comment]
How about leveraging the font stack order to resolve:

`font-family: Lato, -apple-system, 'Segoe UI', Helvetica, Arial, system-ui, sans-serif`

- On macOS using browsers other than Chrome, `-apple-system` is used
- On Windows, `Arial` will be used (making sure it's ordered before `system-ui`)
- On macOS using Chrome, we drop all the way down to `system-ui`, avoiding all the Windows issues we've read about.

**Comment by erinemay** [comment]
Is there any way to preview these changes before making them?

**Comment by ianrohde** [comment]
@erinemay the easiest way is to add this line to the body in browser dev tools:

`font-family: -apple-system, 'Segoe UI', Helvetica, Arial, system-ui, sans-serif;`

SCR-20220824-ifp

**Comment by erinemay** [comment]
un_deploy_block 👍 

Am I correct in assuming there's more discussion happening here? 

>add this line to the body in browser dev tools:
`font-family: -apple-system, 'Segoe UI', Helvetica, Arial, system-ui, sans-serif;`

@davidrans, should we go ahead with some cross browser testing to try this out ⬆️?

**Comment by ianrohde** [comment]
> Am I correct in assuming there's more discussion happening here?

Correct 👍

**Comment by erinemay** [comment]
I'm throwing on the external block label so it's filtered down on our QA Pulldasher list. @davidrans, let us know when you want a full cross browser test of the site. I'll be out next week, but @k0rvusk0r4x is ready to coordinate it as needed.

File: index.json
Comment (federicobadini) [reply index.json:244]: I'm trying to trace back what the problem with `BlinkMacSystemFont` was but I'm struggling a bit because the [issue](https://github.com/iFixit/ifixit/pull/37335) is referencing a Dozuki Slack discussion I've no access to.

Moreover, by skimming through the [chromium issue](https://bugs.chromium.org/p/chromium/issues/detail?id=781344&q=BlinkMacSystemFont&can=1&sort=-modified), I have the sensation that they are discussing something slightly different:

- They are profiling an animation of the `<text>` block inside the SVG
- The performance hit seems to be related to the number of fonts specified, not to using `BlinkMacSystemFont` or not using it.

I've seen that there is a [comment](https://bugs.chromium.org/p/chromium/issues/detail?id=781344#c23) from you @jarstelfox but I'm not seeing any CLS by testing the [react-commerce branch](https://react-commerce-git-use-system-font-ifixit.vercel.app/Parts) with the `BlinkMacSystemFont` font family on chrome.

 Also the [other chromium issue](https://bugs.chromium.org/p/chromium/issues/detail?id=1057654#c35) referenced in #21 seems not to be reproducible anymore.

Can you please give me a little bit more of context?

Comment (jarstelfox) [reply index.json:244]: >Can you please give me a little bit more of context?

💯 

Part of the problem is that the issue we were seeing on the website doesn't seem to be well documented. I searched multiple repos, slack and emails to work back to the exact error.

From my understanding, we were seeing a large reflow during page loads and all of our digging led us to [the chrome bug](https://bugs.chromium.org/p/chromium/issues/detail?id=781344&q=BlinkMacSystemFont&can=1&sort=-modified).

We were not sure this was the error:

>Jarred Stelfox
  19:40
Hey I found it. it’s BlinkMacSystemFont
removing that from the critical css fixes the CLS shifting
https://github.com/tailwindlabs/tailwindcss/issues/1402
^ suggested from here

>Ian Rohde
  19:41
:whoa:
That’s part of the primitive stack. "sansSystem": "-apple-system, BlinkMacSystemFont, 'Segoe UI', Helvetica, Arial, sans-serif", I can make a new version for it later tonight. IIRC we pulled the stack from github, so I’m good with removing it, especially if it’s causing problems

>Jarred Stelfox
  19:48
https://www.google.com/search?q=BlinkMacSystemFont+chrome+bug&oq=BlinkMacSystemFont+[…]g&aqs=chrome..69i57j0i22i30.3051j0j7&sourceid=chrome&ie=UTF-8
A quick google search shows chrome hates that font family
https://bugs.chromium.org/p/chromium/issues/detail?id=1057654#c35
^ found the bug!
Actually, I think it’s: https://bugs.chromium.org/p/chromium/issues/detail?id=781344&q=BlinkMacSystemFont&can=1&sort=-modified
tl;dr: They had a series of rendering bugs that happened a year ago. Now they notice a CLS with it.
One other note: https://bugs.chromium.org/p/chromium/issues/detail?id=554590&q=BlinkMacSystemFont&can=2
It looks like chrome intends to deprecate the font family


I know that thread is not the most helpful, but it seems we were hitting some kind of Layout shifting that went away if we removed the font from the stack. What's tricky is that it's an apple font, so you have to be on a mac only to receive the shifting.

Related, but it looks like the chrome team has put time into fixing bugs with this font overall. Maybe it's safe to use now.

Comment (ianrohde) [reply index.json:244]: `BlinkMacSystemFont` may be safe to use now, but is it reliable long-term? My proposal [here](https://github.com/iFixit/core-primitives/pull/36#issuecomment-1226011159) achieves the same functionality without relying on it.

**Comment by davidrans** [comment]
> That's a little old, but GitHub and Bootstrap didn't add it back it seems :shrug: 

Actually, I was wrong, Bootstrap did add `system-ui` back and dropped `BlinkMacSystemFont`:
https://github.com/twbs/bootstrap/pull/30561
https://github.com/twbs/bootstrap/blob/c3c65911665ab64bdaa15d405db65ee81655dbf3/scss/_variables.scss#L543-L544

File: index.json
Comment (federicobadini) [reply index.json:244]: > `BlinkMacSystemFont` may be safe to use now, but is it reliable long-term? My proposal [here](https://github.com/iFixit/core-primitives/pull/36#issuecomment-1226011159) achieves the same functionality without relying on it.

The linked font stack will make MacOS Chrome render either Helvetica or Arial before SF though

> we were hitting some kind of Layout shifting that went away if we removed the font from the stack. What's tricky is that it's an apple font, so you have to be on a mac only to receive the shifting.

If you have a repro we can definitely test it out to check if it is still an issue

**Comment by federicobadini** [comment]
> > That's a little old, but GitHub and Bootstrap didn't add it back it seems 🤷
> 
> Actually, I was wrong, Bootstrap did add `system-ui` back and dropped `BlinkMacSystemFont`: [twbs/bootstrap#30561](https://github.com/twbs/bootstrap/pull/30561) https://github.com/twbs/bootstrap/blob/c3c65911665ab64bdaa15d405db65ee81655dbf3/scss/_variables.scss#L543-L544

Yet seems that after merge someone still [experienced issues](https://github.com/twbs/bootstrap/pull/30561#issuecomment-638642054) 😅. I've no windows machine available to test that out though

**Comment by ianrohde** [comment]
This stack bypasses the Windows issues seen from using `system-ui` by forcing `Segoe UI` first:

`font-family: Lato, 'Segoe UI', system-ui, Helvetica, Arial, sans-serif`

Windows:
SCR-20220825-i8c

macOS:
SCR-20220825-ifo

I don't see any negatives with this stack, and it doesn't include the issue-prone `BlinkMacSystemFont`.

**Comment by federicobadini** [comment]
> font-family: Lato, 'Segoe UI', system-ui, Helvetica, Arial, sans-serif

Putting `Segoe UI` before `system-ui` should almost completely remove the problem (old windows machines may still have some issues with `system-ui`, not having `Segoe UI` installed by default). 

Seeing the edge cases that we are discussing though I cannot say for sure that this change is not bringing in other issues

**Comment by davidrans** [comment]
I tend to trust GitHub and Bootstrap to both have approaches that work for the vast majority of users. I'd favor copying the stacks of some other trustworthy site/library.

It sounds like we're down to 2 or 3 possible options. Should we turn it over to QA and let them try to hammer each of those with different OS's, locales, browsers and see if they can find any issues?

**Comment by federicobadini** [comment]
> I tend to trust GitHub and Bootstrap to both have approaches that work for the vast majority of users

Agree 💯. 

Clearly if we do experience layout reflows or other issues I agree that we are forced to look for an alternative. 
But if we are not seeing problems I am keen on trusting these players because their sets are probably "a good compromise".

**Comment by davidrans** [comment]
Bootstrap:
```
system-ui, -apple-system, "Segoe UI", Roboto, "Helvetica Neue", "Noto Sans", "Liberation Sans", Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"
```
GitHub:
```
-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji"
```

[Wordpress Admin](https://core.trac.wordpress.org/browser/trunk/src/wp-admin/css/common.css#L220):
```
-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;
```
Medium:
```
-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif
```

I vote we have QA do a thorough test of Bootstrap (i.e. `system-ui`) vs. GitHub (i.e. `BlinkMacSystemFont`) and see if they can find any issues. My guess is that, since both are in such widespread use, we won't find issues with either of them.

If we don't find issues with either of them, then I don't really have a preference. Maybe a very slight lean towards `system-ui` just cause it seems like it's a little more future-oriented?

**Comment by davidrans** [comment]
@ianrohde @federicobadini are we good with choosing two of the options listed above (GitHub and Bootstrap, I'm thinking) and having QA take a pass?

**Comment by ianrohde** [comment]
I also lean towards `system-ui`. The rest are very similar.

**Comment by davidrans** [comment]
Okay, let's QA with GitHub's and Bootstrap's stack and see if we can find any issues with either across OS's/browsers/languages/quantum-portals @erinemay

**Comment by erinemay** [comment]
Adding final steps here. Add each of these stacks to the body of each page.
Bootstrap:

```
font-family: system-ui, -apple-system, "Segoe UI", Roboto, "Helvetica Neue", "Noto Sans", "Liberation Sans", Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"
```

Github:
```
font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji"
```

**Comment by andyg0808** [comment]
dev_block :+1: to keep this from getting CRed or shipped until we've settled on what we want to do here.

**Comment by erinemay** [comment]
High-level findings:
- The fonts chosen looked a little bolder/wider/heavier, causing text to appear a little longer and wrap a little more.
- Most of the time, the additional wrapping was already happening occasionally and it was accommodated well. 
- The few issues we found were all related to the increased wrapping (and they were all pretty minor). 
- From the device/browser combos we tested, each one chose the same font from each stack. There were no visible differences on the page when switching between the two stacks, so either works 🤷 

Issues (all pretty minor):
<details><summary>1. Author badge numbers on answers wrap</summary>

| Lato | system |
| - | - |
| image | image |

</details>


<details><summary>2. In longer languages, the header elements wrap at wider screen-widths</summary>
They already wrap at narrow widths (before going into tablet/mobile mode). They just start to wrap a little wider than before.

| Lato | system |
| - | - |
| image | image |

</details>


<details><summary>3. /Contribute text overriding image on with Lato is exacerbated with system fonts</summary>
Surprise! /Contribute is problematic 🙃
Longer text (long languages) overrides the /Contribute image a little with Lato. Now it's worse.

| Lato | system |
| - | - |
| image | image |

</details>

<details><summary>4. Featured flair icon is smaller</summary>
It looks like it shrinks to accommodate the additional space taken up by the text.

| Lato | system |
| - | - |
| image | image |



</details>

<details><summary>5. View stats wrap more often</summary>
They sometimes already wrap with Lato. Now it's more common.

| Lato | system |
| - | - |
| image | image |

</details>

Details:
- [Spreadsheet to track testing device/browsers](https://docs.google.com/spreadsheets/d/1_b6AfMU5yypX_t9LgU-9DOStTnSHJa6viV_bDTPyM9A/edit#gid=0)
- Device/Browsers
   - MacOS + Safari, FF
   - Windows + Chrome, FF
   - iOS + Safari (mobile and tablet)
   - Android + Chrome (mobile and tablet)
- Languages: French, English, German, Korean 
- We couldn't test mobile FF since we couldn't access the dev-tools on Browserstack.

@davidrans, what's the next step?

**Comment by davidrans** [comment]
I think we should fix all those bugs, but since we're currently loading system fonts in a large percentage of cases I don't think we need to hold up this pull on fixing them.

@federicobadini @ianrohde Since QA couldn't find any issues with either font stack, do either of you have a strong preference. 

I could go either way:
- Most of the sites I looked at use `BlinkMacSystemFont`, including GitHub, which makes me comfortable making that choice.
- On the other hand, `system-ui` seems like the future and there don't seem to be any significant reasons not to use it.

**Comment by ianrohde** [comment]
I want to avoid any potential issues stemming from `BlinkMacSystemFont`. If needed, we can diverge the `lato` and `sansSystem` primitive stacks to suit our preference.

**Comment by federicobadini** [comment]
I do not have a strong preference for one stack over the other. I proposed the Github one but if we feel that we want to avoid `BlinkMacSystemFont` I'm not against choosing the other stack

**Comment by davidrans** [comment]
Okay, I changed the sans font stacks from GitHub -> Bootstrap. I think we can un_dev_block :+1: 

We can probably take a much lighter QA pass just to make sure I didn't typo something.

**Review by ianrohde — APPROVED** [review]
CR ⚪ 

Thanks for the flexibility 👍

**Comment by erinemay** [comment]
Same thing to test? Paste the font stack into the body? What about the other two that are changed? Are we using those anywhere?

**Comment by davidrans** [comment]
Hmm, is there a way to check out this branch of `core-primitives` in the main repo?

**Comment by ianrohde** [comment]
I can do a npm prerelease, but I'm wondering if it would be easier to temporarily adjust the primitives files directly in `node_modules` to match the changes here. It _should_ provide the same thing

**Comment by ianrohde** [comment]
`node_modules` is a no-go. As soon as `make` is run, any manual changes are overwritten. I'll set up a prerelease and report back once published.

**Comment by ianrohde** [comment]
lol `ERROR: Permission to iFixit/core-primitives.git denied to ianrohde.`

**Comment by ianrohde** [comment]
Ready for testing: https://unpkg.com/browse/@core-ds/primitives@2.5.1-rc.0/

**Comment by erinemay** [comment]
> Ready for testing: https://unpkg.com/browse/@core-ds/primitives@2.5.1-rc.0/

How can I apply this to iFixit to test?

**Comment by sterlinghirsh** [comment]
CR :zap:

**Comment by ianrohde** [comment]
> > Ready for testing: https://unpkg.com/browse/@core-ds/primitives@2.5.1-rc.0/
> 
> How can I apply this to iFixit to test?

Sorry about the delay Erin, I've been out sick.

I don't have access to iFixit/ifixit anymore, but you should be able to do a find/replace:

`@core-ds/primitives@2.5.0` to `@core-ds/primitives@2.5.1-rc.0` in package.json (there may be more than one)

**Comment by erinemay** [comment]
Halp! I'm having some trouble with this. I changed all the instances I could find (not just in package.json, which was possibly overkill and maybe my problem?), and after running `make` and `rush update`, this is my diff. 

<details><summary>diff</summary>

```diff
diff --git a/carpenter-frontend/package.json b/carpenter-frontend/package.json
index 4c2b3acc5e2..9d29ba1aa1e 100644
--- a/carpenter-frontend/package.json
+++ b/carpenter-frontend/package.json
@@ -63,7 +63,7 @@
       "@chakra-ui/theme-tools": "~1.3.6",
       "@chakra-ui/toast": "^1.5.7",
       "@core-ds/icons": "^1.3.3",
-      "@core-ds/primitives": "^2.4.5",
+      "@core-ds/primitives": "^2.5.1-rc.0",
       "@emotion/css": "~11.10.0",
       "@emotion/react": "^11.8.1",
       "@emotion/styled": "^11.8.1",
diff --git a/carpenter-js/theme/package.json b/carpenter-js/theme/package.json
index 6a67cbb8a7d..94970c9c96b 100644
--- a/carpenter-js/theme/package.json
+++ b/carpenter-js/theme/package.json
@@ -12,7 +12,7 @@
       "clean": "rimraf ./dist"
    },
    "dependencies": {
-      "@core-ds/primitives": "~2.4.5"
+      "@core-ds/primitives": "~2.5.1-rc.0"
    },
    "devDependencies": {
       "rimraf": "^3.0.2",
diff --git a/common/config/rush/pnpm-lock.yaml b/common/config/rush/pnpm-lock.yaml
index f71810c2de3..b03ef683182 100644
--- a/common/config/rush/pnpm-lock.yaml
+++ b/common/config/rush/pnpm-lock.yaml
@@ -23,7 +23,7 @@ importers:
       '@chakra-ui/theme-tools': ~1.3.6
       '@chakra-ui/toast': ^1.5.7
       '@core-ds/icons': ^1.3.3
-      '@core-ds/primitives': ^2.4.5
+      '@core-ds/primitives': ^2.5.1-rc.0
       '@emotion/css': ~11.10.0
       '@emotion/react': 11.9.0
       '@emotion/styled': 11.8.1
@@ -225,7 +225,7 @@ importers:
       '@chakra-ui/theme-tools': 1.3.6_@chakra-ui+system@1.11.2
       '@chakra-ui/toast': 1.5.9_2lf2su2m7ujhsxzxd2z2brassq
       '@core-ds/icons': 1.3.3_react@16.14.0
-      '@core-ds/primitives': 2.4.5
+      '@core-ds/primitives': 2.5.1-rc.0
       '@emotion/css': 11.10.0_@babel+core@7.17.9
       '@emotion/react': 11.9.0_insruzqejwml3nexkj22dpur3a
       '@emotion/styled': 11.8.1_3gn5s57izm5lhfjxgr4zkvfdti
@@ -461,11 +461,11 @@ importers:
 
   ../../carpenter-js/theme:
     specifiers:
-      '@core-ds/primitives': ~2.4.5
+      '@core-ds/primitives': ~2.5.1-rc.0
       rimraf: ^3.0.2
       typescript: ~4.6.2
     dependencies:
-      '@core-ds/primitives': 2.4.5
+      '@core-ds/primitives': 2.5.1-rc.0
     devDependencies:
       rimraf: 3.0.2
       typescript: 4.6.3
@@ -579,7 +579,7 @@ importers:
       '@babel/preset-env': ^7.14.4
       '@babel/preset-typescript': ^7.13.0
       '@core-ds/icons': ^1.3.0
-      '@core-ds/primitives': ^2.4.5
+      '@core-ds/primitives': ^2.5.1-rc.0
       '@fortawesome/fontawesome-svg-core': ~6.1.1
       '@fortawesome/pro-regular-svg-icons': ~6.1.1
       '@fortawesome/react-fontawesome': ~0.1.18
@@ -619,7 +619,7 @@ importers:
       typescript: ^3.9.9
     dependencies:
       '@core-ds/icons': 1.3.3_react@16.14.0
-      '@core-ds/primitives': 2.4.5
+      '@core-ds/primitives': 2.5.1-rc.0
       '@fortawesome/fontawesome-svg-core': 6.1.1
       '@fortawesome/pro-regular-svg-icons': 6.1.1
       '@fortawesome/react-fontawesome': 0.1.18_3o7rhq5e2fd3auzno3wstxs4ey
@@ -673,7 +673,7 @@ importers:
       '@chakra-ui/icons': ^1.0.15
       '@chakra-ui/react': ^1.6.6
       '@chakra-ui/system': ~2.1.1
-      '@core-ds/primitives': ^1.5.0
+      '@core-ds/primitives': ^2.5.1-rc.0
       '@emotion/react': 11.9.0
       '@emotion/styled': 11.8.1
       '@ifixit/react-components': ^0.3.0
@@ -703,7 +703,7 @@ importers:
       '@chakra-ui/icons': 1.1.7_43khzrqpf4s2psydq72xkc5sca
       '@chakra-ui/react': 1.8.8_xkgxhw3wemj3pp6vzlnnjihwla
       '@chakra-ui/system': 2.1.1_bgqmsvm4hz6izcmpcwescmz73y
-      '@core-ds/primitives': 1.5.0
+      '@core-ds/primitives': 2.5.1-rc.0
       '@emotion/react': 11.9.0_426xlbpu4clsxae4qx32qwknde
       '@emotion/styled': 11.8.1_neh35hrqppbvdgjxxnlc3q5icm
       '@ifixit/react-components': 0.3.0_lzomchjozpjp3t5kmo7yr7pgeq
@@ -739,7 +739,7 @@ importers:
       '@babel/runtime': ~7.16.7
       '@chakra-ui/react': ~1.8.6
       '@core-ds/icons': ^1.0.0
-      '@core-ds/primitives': ^2.4.5
+      '@core-ds/primitives': ^2.5.1-rc.0
       '@emotion/react': 11.9.0
       '@emotion/styled': 11.8.1
       '@fortawesome/fontawesome-svg-core': ~6.1.0
@@ -770,7 +770,7 @@ importers:
       '@babel/runtime': 7.16.7
       '@chakra-ui/react': 1.8.8_ivoplvjmqgmmcxll5nqusjf7b4
       '@core-ds/icons': 1.3.3_react@16.14.0
-      '@core-ds/primitives': 2.4.5
+      '@core-ds/primitives': 2.5.1-rc.0
       '@emotion/react': 11.9.0_dyau445mh2pdagbkb2hgesg4xi
       '@emotion/styled': 11.8.1_kbp7jgi2k3acql37g5mmrjdsjy
       '@fortawesome/fontawesome-svg-core': 6.1.1
@@ -848,12 +848,12 @@ importers:
   ../../runtime_deps_install:
     specifiers:
       '@core-ds/icons': ^1.3.3
-      '@core-ds/primitives': 2.4.5
+      '@core-ds/primitives': 2.5.1-rc.0
       imagemarkup: iFixit/node-markup.git#f295d757306141d4658a08d2ebb2475887b52f08
       react: 16.14.0
     dependencies:
       '@core-ds/icons': 1.3.3_react@16.14.0
-      '@core-ds/primitives': 2.4.5
+      '@core-ds/primitives': 2.5.1-rc.0
       imagemarkup: github.com/iFixit/node-markup/f295d757306141d4658a08d2ebb2475887b52f08
     devDependencies:
       react: 16.14.0
@@ -6309,12 +6309,8 @@ packages:
       react: 16.14.0
     dev: false
 
-  /@core-ds/primitives/1.5.0:
-    resolution: {integrity: sha512-wOd/FZaVW1laDbY6D63a80xCagmhiqpTgM0/bEJEOhGz39DCBDZUdUmxrWc2N/PfgEFf2CMu3JPiIy0S6GZPHQ==}
-    dev: false
-
-  /@core-ds/primitives/2.4.5:
-    resolution: {integrity: sha512-0gUTtOrGUJdmCRRIQ8JDhNNZz6tf2A9HCouJxfT7aTh3dIIYTTEv5JlVjtVn6Yjtt2sEOMer/7A3vQcWdRQetQ==}
+  /@core-ds/primitives/2.5.1-rc.0:
+    resolution: {integrity: sha512-VuAqx+ow3Pb01r/ggvdBjSNkJ+NE2Cl+5smm9BPRCYvc1nVNiERvQso/TQRKy5MI14W/MgFbeI+WaGlHPzl8yw==}
     dev: false
 
   /@cspotcode/source-map-support/0.8.1:
diff --git a/common/config/rush/repo-state.json b/common/config/rush/repo-state.json
index 4762d7b456f..c9b3b3729a1 100644
--- a/common/config/rush/repo-state.json
+++ b/common/config/rush/repo-state.json
@@ -1,5 +1,5 @@
 // DO NOT MODIFY THIS FILE MANUALLY BUT DO COMMIT IT. It is generated and used by Rush.
 {
-  "pnpmShrinkwrapHash": "db6cd466fb0b857a57106e0a1fd0d1f16b03c655",
+  "pnpmShrinkwrapHash": "3df2accb6d47b2213b760792118605d08a1866c2",
   "preferredVersionsHash": "f21e10d7a60116c88059eb278beb50276307d038"
 }
diff --git a/device-picker/package.json b/device-picker/package.json
index a9dba869548..1ebc3a24e3a 100644
--- a/device-picker/package.json
+++ b/device-picker/package.json
@@ -28,7 +28,7 @@
    },
    "dependencies": {
       "@core-ds/icons": "^1.3.0",
-      "@core-ds/primitives": "^2.4.5",
+      "@core-ds/primitives": "^2.5.1-rc.0",
       "@fortawesome/fontawesome-svg-core": "~6.1.1",
       "@fortawesome/pro-regular-svg-icons": "~6.1.1",
       "@fortawesome/react-fontawesome": "~0.1.18",
diff --git a/nextjs-community/package.json b/nextjs-community/package.json
index f6246bc744f..e818ad6e54c 100644
--- a/nextjs-community/package.json
+++ b/nextjs-community/package.json
@@ -13,7 +13,7 @@
       "@chakra-ui/icons": "^1.0.15",
       "@chakra-ui/react": "^1.6.6",
       "@chakra-ui/system": "~2.1.1",
-      "@core-ds/primitives": "^1.5.0",
+      "@core-ds/primitives": "^2.5.1-rc.0",
       "@emotion/react": "^11",
       "@emotion/styled": "^11",
       "@ifixit/react-components": "^0.3.0",
diff --git a/product-page-components/package.json b/product-page-components/package.json
index 660e3cdc310..564a7a02049 100644
--- a/product-page-components/package.json
+++ b/product-page-components/package.json
@@ -36,7 +36,7 @@
       "@babel/runtime": "~7.16.7",
       "@chakra-ui/react": "~1.8.6",
       "@core-ds/icons": "^1.0.0",
-      "@core-ds/primitives": "^2.4.5",
+      "@core-ds/primitives": "^2.5.1-rc.0",
       "@emotion/react": "^11",
       "@emotion/styled": "^11",
       "@fortawesome/fontawesome-svg-core": "~6.1.0",
diff --git a/react-commerce b/react-commerce
--- a/react-commerce
+++ b/react-commerce
@@ -1 +1 @@
-Subproject commit 4cdb90991ba5479924aa7d9a7db0800f55b919b5
+Subproject commit 4cdb90991ba5479924aa7d9a7db0800f55b919b5-dirty
diff --git a/runtime_deps_install/package.json b/runtime_deps_install/package.json
index 7b2fbd40c1d..0f090801ced 100644
--- a/runtime_deps_install/package.json
+++ b/runtime_deps_install/package.json
@@ -7,7 +7,7 @@
    },
    "dependencies": {
       "@core-ds/icons": "^1.3.3",
-      "@core-ds/primitives": "2.4.5",
+      "@core-ds/primitives": "2.5.1-rc.0",
       "imagemarkup": "iFixit/node-markup.git#f295d757306141d4658a08d2ebb2475887b52f08"
    },
    "devDependencies": {


```

</details>

And I'm still seeing the unchanged font stack:
image

**Comment by ianrohde** [comment]
@erinemay, the change should only be made in package.json files. It's possible you are running an older version of primitives. Try a search for `@core-ds/primitives@2.` scoped to `package.json` and replace the whole version string with `@core-ds/primitives@2.5.1-rc.0`.

Past that, there's little I can do to help without access to the repo. 😞

**Comment by erinemay** [comment]
Thank you so much @ianrohde and @andyg0808! It worked!

I'm seeing the new Lato family on cominor pages, identical to the Bootstrap stack. Depending on the browser, walking down the list and deleting fonts sets each next browser-appropriate font as the font on the page. 

```
font-family: Lato, system-ui, -apple-system, 'Segoe UI', Roboto, 'Helvetica Neue', 'Noto Sans', 'Liberation Sans', Arial, sans-serif;
```

@davidrans, where do we use the monoSystem and sansSystem font families?

**Comment by davidrans** [comment]
We use them in this pull: https://github.com/iFixit/ifixit/pull/41227

**Comment by erinemay** [comment]
> We use them in this pull: [iFixit/ifixit#41227](https://github.com/iFixit/ifixit/pull/41227)

Hmm, I'm still seeing Lato on that branch. It's the correctly updated version from here when I use  `2.5.1-rc.0`, but it's still Lato. Should it be completely gone? 
image

**Comment by ianrohde** [comment]
I can almost guarantee that the lato primitive stack is still being used. If you want to switch iFixit to the system stack, it will take more code changes. I'm not sure if those changes were intended to be a part of this PR.

**Comment by davidrans** [comment]
Interestingly, GitHub just updated their font stack: https://github.com/orgs/community/discussions/39226

File: index.json
Comment (davidrans) [inline index.json:244]: This looks like it matches GitHub:
image

File: index.json
Comment (davidrans) [inline index.json:245]: This looks like it matches GitHub:
image

**Pull request merged**
